src/Platform/Microsoft.Testing.Platform/Requests/RequestNotSupportedException.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Testing.Platform.Requests;
+
+[Serializable]
```

Comment:
I think we should avoid this one based on docs updates: https://github.com/dotnet/docs/issues/34893